packages/field-plugin/src/createFieldPlugin/createFieldPlugin.ts

```diff
@@ -85,24 +88,31 @@ export const createFieldPlugin: CreateFieldPlugin = ({
     Exclude<typeof validateContent, undefined>
   >['content']
 
-  const { actions, messageCallbacks, onHeightChange, initialize } =
-    createPluginActions<InferredContent>({
-      uid,
-      postToContainer,
-      onUpdateState: (data) => {
-        onUpdateState({
-          type: 'loaded',
-          data,
-          actions,
-        })
-      },
-      validateContent:
-        validateContent ||
-        ((content) => ({ content: content as InferredContent })),
-    })
+  const {
+    actions,
+    messageCallbacks,
+    onHeightChange,
+    onKeydownEsc,
+    initialize,
+  } = createPluginActions<InferredContent>({
+    uid,
+    postToContainer,
+    onUpdateState: (data) => {
+      onUpdateState({
+        type: 'loaded',
+        data,
+        actions,
+      })
+    },
+    validateContent:
+      validateContent ||
+      ((content) => ({ content: content as InferredContent })),
+  })
 
   const cleanupHeightChangeListener = createHeightChangeListener(onHeightChange)
 
+  const cleanupKeydownEscListener = createKeydownEscListener(onKeydownEsc)
```

Comment:
The "action" `onKeydownEsc` might be redundant, because you simply could call `actions.setModalOpen(false)`.

Reply:
I used the same structure as per `onHeightChange()` here, first to be consistent and second because having the method defined in the `CreatePluginActions` makes easier to find later (every method implementation is already there), here we just have the invokes.

Here you can see the implementation: https://github.com/storyblok/field-plugin/pull/404/files/10441a2ab847700dd16be9fe7dc2eaca0065a09b#diff-e62bdc8d2bfaac8dcd9c0ec5f420b868131bc32f10c4be5743ba3e5e28ebe938R90-R93, which also have a discussion (marked as resolved) on why this approach and not a Promise was the choice 🙏